cloudinit/analyze/__main__.py

```diff
@@ -192,6 +192,10 @@ def analyze_boot(name, args):
     }
 
     outfh.write(status_map[status_code].format(**kwargs))
+    if outfh != sys.stdout:
```

Comment:
I think that before merging this as-is, we should at least approach the return from `configure_io` consistently. If you'd rather not deal with that right now, I think I'd prefer we remove this section before merging.

Reply:
The conditional exists because the sys file descriptors are provided by the sys module and therefore I don't think we should close them, but if outfh/infh are allocated by us, we should clean them up.

What exactly do you mean by "approach the return from `configure_io` consistently?

Reply:
Sorry, just that the function is called a few times in this file, but we're only addressing it here.

Reply:
@TheRealFalcon Agreed, thanks! I just pushed a change addressing this comment.